gateway-discovery-cm/src/test/java/org/apache/knox/gateway/topology/discovery/cm/ClouderaManagerServiceDiscoveryTest.java

```diff
@@ -331,6 +331,17 @@ public void testLivyDiscoverySSL() {
     assertEquals("https://" + hostName + ":" + port, livyURLs.get(0));
   }
 
+  @Test
+  public void testLivyDiscoveryWhenTransportModeIsBinary() {
+    final String hostName    = "livy-host";
+    final String port        = "8998";
+    ServiceDiscovery.Cluster cluster = doTestLivyDiscovery(hostName, port, false, "binary");
```

Comment:
This doesn't make much sense to me. Livy discovery should be on the HTTP endpoint. Binary vs HTTP would be on the thrift server not on the HTTP endpoint....

Reply:
Ok